src/porepy/models/contact_mechanics_model.py

```diff
@@ -399,6 +399,43 @@ def reconstruct_local_displacement_jump(self, data_edge, from_iterate=True):
         u_mortar_local = project_to_local * displacement_jump_global_coord
         return u_mortar_local.reshape((self.Nd, -1), order="F")
 
+    def reconstruct_stress(self) -> None:
+        """
+        Compute the stress in the highest-dimensional grid based on the displacement
+        states in that grid, adjacent interfaces and global boundary conditions.
+        
+        The stress is stored in the data dictionary of the highest dimensional grid,
+        in [pp.STATE]['stress'].
+        
+        """
+        g = self._nd_grid()
+        d = self.gb.node_props(g)
+
+        mpsa = pp.Mpsa(self.mechanics_parameter_key)
+
+        u = d[pp.STATE][self.displacement_variable]
```

Comment:
I think it would be useful to include the option to compute from previous iterate. Could e.g. be a parameter of the method